docs/integration-guide.md

```diff
@@ -107,6 +107,342 @@ Here's how we recommend doing it:
+data allows you to do two things:
+
+1. Determine how you will need to structure your webook code, including what
+   message types your integration should support and how; and,
+2. Create fixtures for your webook tests.
+
+Fixtures enable the testing of webhook integration code without the need to
+actually contact the service being integrated.
+
+Because `Hello World` is a very simple webhook that does one thing, it requires
+only one fixture, `zerver/fixtures/helloworld/helloworld_hello.json`:
+
+```
+{
+  "featured_title":"Marilyn Monroe",
+  "featured_url":"https://en.wikipedia.org/wiki/Marilyn_Monroe",
+}
+```
+
+When writing your own webhook integration, you'll want to write a test function
+for each distinct message condition your webhook supports. You'll also need a
+corresponding fixture for each of these tests. See [Step 3: Create
+tests](#step-3-create-tests) or [Testing](testing.html) for further details.
+
+#### Step 1: Create main webhook code
+
+The majority of the code for your webhook integration will be in a single
+python file in `zerver/views/webhooks/`.
+
+The Hello World integration is in `zerver/views/webhooks/helloworld.py`:
+
+```
+from __future__ import absolute_import
+from django.utils.translation import ugettext as _
+from zerver.lib.actions import check_send_message
+from zerver.lib.response import json_success, json_error
+from zerver.decorator import REQ, has_request_variables, api_key_only_webhook_view
+from zerver.lib.validator import check_dict, check_string
+
+@api_key_only_webhook_view('HelloWorld')
+@has_request_variables
+def api_helloworld_webhook(request, user_profile, client,
+                           payload=REQ(argument_type='body'),
+                           stream=REQ(default='test'),
+                           topic=REQ(default='Hello World')):
+
+  # construct the body of the message
+  body = 'Hello! I am happy to be here! :smile: '
+
+  # try to add the Wikipedia article of the day
+  # return appropriate error if not successful
+  try:
+      body_template = '\nThe Wikipedia featured article for today is **[{featured_title}]({featured_url})**'
+      body += body_template.format(**payload)
+  except KeyError as e:
+      return json_error(_("Missing key {} in JSON").format(e.message))
+
+  # send the message
+  check_send_message(user_profile, client, 'stream', [stream], topic, body)
+
+  # return json result
+  return json_success()
+
+```
+
+The above code imports the required functions and defines the main webhook
+function `api_helloworld_webook`, decorating it with `api_key_only_webhook_view` and
+`has_request_variables`.
+
+You must pass the name of your webhook to the `api_key_only_webhook_view`
+decorator. Here we have used `HelloWorld`.
+
+You may name your webhook function whatever you like, though it's a good idea
+to be consistent with other webhook integrations. We recommend following the
+format `api_webhookname_webhook`.
```

Comment:
Actually, this is becoming semi-standardized in https://github.com/zulip/zulip/pull/1401, so we should probably more firmly recommend calling their function that (with `webhookname` all in lower-case).  That PR will also slightly change these instructions, but only slightly (replace the `urls.py` step with adding an entry in `zerver/lib/integrations.py`)